ban-cooking.lua

```diff
@@ -0,0 +1,433 @@
+-- convenient way to ban cooking categories of food
+-- based on ban-cooking.rb by Putnam: https://github.com/DFHack/scripts/pull/427/files
+-- Putnams work completed by TBSTeun
+
+local kitchen = df.global.ui.kitchen
+
+local already_banned = {}
+
+local function make_key(mat_type, mat_index, type, subtype)
+    return ('%s:%s:%s:%s'):format(mat_type, mat_index, type, subtype)
+end
+
+local function reaction_product_id_contains(reaction_product, str)
+    for _, s in ipairs(reaction_product.id) do
+        if s.value == str then
+            return true
+        end
+    end
+
+    return false
+end
+
+local function ban_cooking(print_name, mat_type, mat_index, type, subtype)
+    local key = make_key(mat_type, mat_index, type, subtype)
+    -- Skip adding a new entry further below, if the item is already banned.
+    if already_banned[key] then
+        return
+    end
+    -- The item hasn't already been banned, so we do that here by appending its values to the various arrays
+    print(print_name .. ' has been banned!')
+    kitchen.mat_types:insert('#', mat_type)
+    kitchen.mat_indices:insert('#', mat_index)
+    kitchen.item_types:insert('#', type)
+    kitchen.item_subtypes:insert('#', subtype)
+    kitchen.exc_types:insert('#', df.kitchen_exc_type.Cook)
+
+    already_banned[key] = {}
+    already_banned[key].isBanned = true
```

Comment:
the `isBanned` value is never used (and is always `true`) as far as I can see. is there a reason you included it?